File: src/main/java/com/kingg/api_vacunas_panama/events/AuthenticationSuccessListener.java
package com.kingg.api_vacunas_panama.events;

import com.kingg.api_vacunas_panama.service.UsuarioTransactionService;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.context.ApplicationListener;
import org.springframework.security.authentication.event.AuthenticationSuccessEvent;
import org.springframework.stereotype.Component;

import java.util.UUID;

@Slf4j
@Component
@RequiredArgsConstructor
public class AuthenticationSuccessListener implements ApplicationListener<AuthenticationSuccessEvent> {
    private final UsuarioTransactionService usuarioTransactionService;

    @Override
    public void onApplicationEvent(AuthenticationSuccessEvent event) {
        log.debug("Authentication Success Event: {}", event.toString());
        if (event.getAuthentication() != null && event.getAuthentication().isAuthenticated()) {
            String userId = event.getAuthentication().getName();
            log.debug("User ID for update last used: {}", userId);
            if (userId != null && !userId.isBlank()) {
                usuarioTransactionService.updateLastUsed(UUID.fromString(userId));
            }
        }
    }

}

File: src/main/kotlin/io/github/kingg22/api/vacunas/panama/modules/usuario/events/AuthenticationSuccessListener.kt
package io.github.kingg22.api.vacunas.panama.modules.usuario.events

/*
TODO find a solution to this

@Component
class AuthenticationSuccessListener(private val usuarioService: UsuarioService) :
    ApplicationListener<AuthenticationSuccessEvent> {
    private val log = logger()

    override fun onApplicationEvent(event: AuthenticationSuccessEvent) {
        log.debug("Authentication Success Event: {}", event.toString())
        if (event.authentication != null && event.authentication.isAuthenticated) {
            val userId = event.authentication.name
            log.debug("User ID for update last used: {}", userId)
            if (!userId.isNullOrBlank()) {
                try {
                    val uuid: UUID = UUID.fromString(userId)
                    CoroutineScope(Dispatchers.Default).launch {
                        usuarioService.updateLastUsed(uuid)
                        log.debug("User updated successfully for usuario: {}", userId)
                    }
                } catch (e: IllegalArgumentException) {
                    log.error("Invalid user ID: {} is not a valid UUID", userId, e)
                }
            }
        }
    }
}
*/
